### PR TITLE
[Prisma Cloud] Initial Release for Prisma Cloud

### PR DESCRIPTION
**Pull request by mohitjha-elastic**
## What does this PR do?

- Generated the skeleton of the Prisma Cloud integration package.
- Added data stream.
- Added data collection logic for the data stream.
- Added the ingest pipeline for the data stream.
- Mapped fields according to the ECS schema and added Fields metadata in the appropriate yml files.
- Added test for pipeline for the data stream.

## Integration release checklist
This checklist is intended for integrations maintainers to ensure consistency when creating or updating a Package, Module or Dataset for an Integration.

### All changes
- [x] Change follows the [contributing guidelines](https://github.com/elastic/integrations/blob/master/CONTRIBUTING.md)
- [x] Supported versions of the monitoring target is documented
- [x] Supported operating systems are documented (if applicable)
- [x] Integration or [System tests](https://github.com/elastic/elastic-package/blob/master/docs/howto/system_testing.md) exist
- [x] Documentation exists
- [x] Fields follow [ECS](https://github.com/elastic/ecs) and [naming conventions](https://www.elastic.co/guide/en/beats/devguide/master/event-conventions.html)
- [x] At least a manual test with ES / Kibana / Agent has been performed.
- [x] Required Kibana version set to: ^8.8.0

### New Package
- [x] Screenshot of the "Add Integration" page on Fleet added

### Log dataset changes
- [x] [Pipeline tests](https://github.com/elastic/elastic-package/blob/master/docs/howto/pipeline_testing.md) exist (if applicable)
- [x] Generated output for at least 1 log file exists
- [x] Sample event (sample_event.json) exists

## How to test this PR locally

- Clone integrations repo.
- Install elastic package locally.
- Start elastic stack using elastic-package.
- Move to integrations/packages/prisma_cloud directory.
- Run the following command to run tests. 
> elastic-package test 

## Related issues

- Closes https://github.com/elastic/integrations/issues/1132

## Automated Test
[test-file.txt](https://github.com/elastic/integrations/files/12843447/test-file.txt)

## Screenshot
image
image

We are facing message size exceeding errors so need to validate it in the cloud instance.
image

### CODE REVIEW

File: packages/prisma_cloud/data_stream/audit/agent/stream/input.yml.hbs [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        "application/json",
+        {"username":state.user,"password":state.password}.encode_json()
+      ).do_request().as(resp, bytes(resp.Body).decode_json().as(body, {
+          "access_token": body.token,
+      }))
+    ).as(state, state.with(
+      request("GET",
+        state.url +
+        "/audit/redlock?timeType=relative&timeAmount=" +
+        (has(state.cursor) && has(state.cursor.time_amount_in_minute) && state.cursor.time_amount_in_minute != null
```
Comment (kcreddy) [inline packages/prisma_cloud/data_stream/audit/agent/stream/input.yml.hbs:39]: Doesn't `has(state.cursor.time_amount_in_minute)` and `state.cursor.time_amount_in_minute != null` both mean the same thing?
https://github.com/google/cel-spec/blob/master/doc/langdef.md#field-selection

File: packages/prisma_cloud/data_stream/alert/_dev/test/system/test-input-config.yml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+input: cel
+service: prisma_cloud
+vars:
+  username: xxxx
+  password: xxxx
+data_stream:
+  vars:
+    url: http://{{Hostname}}:{{Port}}
+    preserve_original_event: true
+    preserve_duplicate_custom_fields: true
```
Comment (kcreddy) [inline packages/prisma_cloud/data_stream/alert/_dev/test/system/test-input-config.yml:10]: Can you add `assert.hit_count` to all system test configs?

File: packages/prisma_cloud/data_stream/audit/agent/stream/input.yml.hbs (hunk 1 above)
Comment (mohitjha-elastic) [reply packages/prisma_cloud/data_stream/audit/agent/stream/input.yml.hbs:39]: It's not working in our case as an individual condition from the above is unable to check if the field is present and not null hence throwing an error.

Comment (kcreddy) [reply packages/prisma_cloud/data_stream/audit/agent/stream/input.yml.hbs:39]: Interesting. Thanks for testing.

**Review by piyush-elastic — APPROVED** [review]

File: packages/prisma_cloud/data_stream/host/_dev/test/system/test-input-config.yml [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+vars:
+  username: xxxx
+  password: xxxx
+data_stream:
+  vars:
+    url: http://{{Hostname}}:{{Port}}
+    preserve_original_event: true
+    preserve_duplicate_custom_fields: true
+assert:
+  hit_count: 500
```
Comment (kcreddy) [inline packages/prisma_cloud/data_stream/host/_dev/test/system/test-input-config.yml:12]: How are we getting 500 events? Where are these 500 events defined?

**Comment by leandrojmp** [comment]
Hello, just curious, is this integration planned for 8.10 or just 8.11?

**Comment by mohitjha-elastic** [comment]
> Hello, just curious, is this integration planned for 8.10 or just 8.11?

Hey @leandrojmp, We have planned it for 8.10.1.

File: packages/prisma_cloud/data_stream/host/agent/stream/input.yml.hbs [hunk 4]
```diff
@@ -55,13 +56,29 @@ program: |
                 []
             ),
             "url": state.url,
-            "new_offset": has(state.new_offset) ? string(int(state.new_offset) + int(state.batch_size)) : string(int(state.offset) + int(state.batch_size)),
-            "want_more": inner_body != null && (int(state.batch_size) > 0),
+            "want_more": inner_body != null && inner_body.size() > 0,
             "user": state.user,
             "password": state.password,
             "batch_size": string(state.batch_size),
+            "access_token": state.access_token,
```
Comment (kcreddy) [inline packages/prisma_cloud/data_stream/host/agent/stream/input.yml.hbs:63]: you could save this token in redact config

**Review by kcreddy — APPROVED** [review]
LGTM

**Comment by leandrojmp** [comment]
Hello, 

We are planning to test this integration, but one question, is the Incident Audit dataset only available when using TCP/UDP? The datastreams for this dataset only have those two inputs available.

Not sure if this is a limitation of the Prisma Cloud tool or the integration.

**Pull request merged**